extras/rya.streams/api/src/main/java/org/apache/rya/streams/api/queries/QueryRepository.java

```diff
@@ -44,8 +48,9 @@
      *   otherwise {@code false}.
      * @return The {@link StreamsQuery} used in Rya Streams.
      * @throws QueryRepositoryException Could not add the query.
+     * @throws IllegalStateException The Service has not been started, but has been subscribed to.
      */
-    public StreamsQuery add(final String query, boolean isActive) throws QueryRepositoryException;
+    public StreamsQuery add(final String query, boolean isActive) throws QueryRepositoryException, IllegalStateException;
```

Comment:
Add tests for these conditions. I don't see any edits to the code in this commit that implements this interface change.